example/advection/advection_package.cpp

```diff
@@ -205,51 +204,49 @@ Real EstimateTimestep(Container<Real> &rc) {
 // This routine implements all the "physics" in this example
 TaskStatus CalculateFluxes(Container<Real> &rc) {
   MeshBlock *pmb = rc.pmy_block;
-  int is = pmb->is;
-  int js = pmb->js;
-  int ks = pmb->ks;
-  int ie = pmb->ie;
-  int je = pmb->je;
-  int ke = pmb->ke;
+  IndexRange ib = pmb->cellbounds.GetBoundsI(IndexDomain::interior);
+  IndexRange jb = pmb->cellbounds.GetBoundsJ(IndexDomain::interior);
+  IndexRange kb = pmb->cellbounds.GetBoundsK(IndexDomain::interior);
+  int ncellsi = pmb->cellbounds.ncellsi(IndexDomain::entire);
 
   CellVariable<Real> &q = rc.Get("advected");
   auto pkg = pmb->packages["advection_package"];
   const auto &vx = pkg->Param<Real>("vx");
   const auto &vy = pkg->Param<Real>("vy");
   const auto &vz = pkg->Param<Real>("vz");
 
-  ParArrayND<Real> ql("ql", q.GetDim(4), pmb->ncells1);
-  ParArrayND<Real> qr("qr", q.GetDim(4), pmb->ncells1);
-  ParArrayND<Real> qltemp("qltemp", q.GetDim(4), pmb->ncells1);
+  ParArrayND<Real> ql("ql", q.GetDim(4), ncellsi);
+  ParArrayND<Real> qr("qr", q.GetDim(4), ncellsi);
+  ParArrayND<Real> qltemp("qltemp", q.GetDim(4), ncellsi);
 
   // get x-fluxes
-  for (int k = ks; k <= ke; k++) {
-    for (int j = js; j <= je; j++) {
+  for (int k = kb.s; k <= kb.e; k++) {
+    for (int j = jb.s; j <= jb.e; j++) {
       // get reconstructed state on faces
-      pmb->precon->DonorCellX1(k, j, is - 1, ie + 1, q.data, ql, qr);
+      pmb->precon->DonorCellX1(k, j, ib.s - 1, ib.e + 1, q.data, ql, qr);
       if (vx > 0.0) {
-        for (int i = is; i <= ie + 1; i++) {
+        for (int i = ib.s; i <= ib.e + 1; i++) {
           q.flux[X1DIR](k, j, i) = ql(i) * vx;
         }
       } else {
-        for (int i = is; i <= ie + 1; i++) {
+        for (int i = ib.s; i <= ib.e + 1; i++) {
           q.flux[X1DIR](k, j, i) = qr(i) * vx;
         }
       }
     }
   }
   // get y-fluxes
   if (pmb->pmy_mesh->ndim >= 2) {
-    for (int k = ks; k <= ke; k++) {
-      pmb->precon->DonorCellX2(k, js - 1, is, ie, q.data, ql, qr);
-      for (int j = js; j <= je + 1; j++) {
-        pmb->precon->DonorCellX2(k, j, is, ie, q.data, qltemp, qr);
+    for (int k = kb.s; k <= kb.e; k++) {
+      pmb->precon->DonorCellX2(k, jb.s - 1, ib.s, ib.e, q.data, ql, qr);
+      for (int j = jb.s; j <= jb.e + 1; j++) {
+        pmb->precon->DonorCellX2(k, j, ib.s, ib.e, q.data, qltemp, qr);
```

Comment:
Obviously beyond the scope of your current MR, but we can probably pass in bounds objects themselves, instead of the bounds indexes from now on.

Reply:
Exactly!